matching_brackets.exs

```diff
@@ -0,0 +1,49 @@
+defmodule BracketsBalance do
+  @brackets %{
+    "{" => "}",
+    "(" => ")",
+    "[" => "]"
+  }
+
+  def add_to_stack(stack, counter, check_string) do
+    if counter < length(check_string) do
+      stack =
+        if Map.has_key?(@brackets, Enum.at(check_string, counter)) do
+          stack ++ [Enum.at(check_string, counter)]
```

Comment:
You would have much better time if you placed items into stack in reversed order - most recent is first, and oldest is last. Then here you need to insert into first element, shifting rest to the right:
```
[Enum.at(check_string, counter) | stack]
```